### PR TITLE
Improve Rsyslog rules to support Rainer script syntax

### PR DESCRIPTION
**Pull request by jan-cerny**
#### Description:
Improve OVAL checks so that the rule will pass if the modern Rainer script syntax is used. Up until now these rules passed only when the legacy configuration is used.

This changes OVAL in the following rules:

- rsyslog_cron_logging
- rsyslog_encrypt_offload_actionsendstreamdriverauthmode
- rsyslog_encrypt_offload_actionsendstreamdrivermode
- rsyslog_encrypt_offload_defaultnetstreamdriver

For more details, please read commit messages of all commits.

#### Rationale:
Fixes: https://issues.redhat.com/browse/RHEL-1816

### CODE REVIEW

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml [hunk 1]
```diff
@@ -26,6 +30,21 @@
...
+                                id="test_{{{rule_id}}}_action_send_stream_driver_auth_mode_rainer" version="1">
+
+        <ind:object object_ref="obj_{{{rule_id}}}_action_send_stream_driver_auth_mode_rainer" />
+    </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_object id="obj_{{{rule_id}}}_action_send_stream_driver_auth_mode_rainer"
+                                    comment="Check if StreamDriverAuthMode is set to x509/name in /etc/rsyslog.conf using Rainer script"
+                                    version="1">
+        <ind:filepath>/etc/rsyslog.conf</ind:filepath>
+        <ind:pattern operation="pattern match">^\s*action\(.*(?i)\btype\b(?-i)="omfwd".*(?i)\bStreamDriverAuthMode\b(?-i)="x509/name".*\)\s*$</ind:pattern>
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml:44]: Could you also update the rule description for RainerScript syntax, please?

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml [hunk 2]
```diff
@@ -40,4 +59,20 @@
...
+                                id="test_{{{rule_id}}}_action_send_stream_driver_auth_mode_dir_rainer" version="1">
+        <ind:object object_ref="obj_{{{rule_id}}}_action_send_stream_driver_auth_mode_dir_rainer" />
+    </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_object id="obj_{{{rule_id}}}_action_send_stream_driver_auth_mode_dir_rainer"
+                                    comment="Check if StreamDriverAuthMode is set to x509/name in files in /etc/rsyslog.d using Rainer script"
+                                    version="1">
+        <ind:path>/etc/rsyslog.d</ind:path>
+        <ind:filename operation="pattern match">^.*conf$</ind:filename>
+        <ind:pattern operation="pattern match">^\s*action\(.*(?i)\btype\b(?-i)="omfwd".*(?i)\bStreamDriverAuthMode\b(?-i)="x509/name".*\)\s*$</ind:pattern>
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml:74]: Same regex already used before. Could you consider to create a variable, please?

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml [hunk 3]
```diff
@@ -41,4 +60,19 @@
...
+                                id="test_{{{rule_id}}}_action_send_stream_driver_mode_rsyslog_dir_rainer" version="1">
+        <ind:object object_ref="obj_{{{rule_id}}}_action_send_stream_driver_mode_rsyslog_dir_rainer" />
+    </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_object id="obj_{{{rule_id}}}_action_send_stream_driver_mode_rsyslog_dir_rainer"
+                                    comment="Check if StreamDriverMode is set to 1 in files in /etc/rsyslog.d using Rainer script"
+                                    version="1">
+        <ind:path>/etc/rsyslog.d</ind:path>
+        <ind:filename operation="pattern match">^.*conf$</ind:filename>
+        <ind:pattern operation="pattern match">^\s*action\(.*(?i)\btype\b(?-i)="omfwd".*(?i)\bStreamDriverMode\b(?-i)="1".*\)\s*$</ind:pattern>
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml:75]: Same regex used before. Could you consider to create a variable, please?

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml [hunk 4]
```diff
@@ -27,6 +31,21 @@
...
+                                id="test_{{{rule_id}}}_default_netstream_rsyslog_rainer" version="1">
+
+        <ind:object object_ref="obj_{{{rule_id}}}_default_netstream_rsyslog_rainer" />
+    </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_object id="obj_{{{rule_id}}}_default_netstream_rsyslog_rainer"
+                                    comment="Check if DefaultNetstreamDriver is set to gtls in /etc/rsyslog.conf using Rainer script"
+                                    version="1">
+        <ind:filepath>/etc/rsyslog.conf</ind:filepath>
+        <ind:pattern operation="pattern match">^\s*global\(.*(?i)\bDefaultNetStreamDriver\b(?-i)="gtls".*\)\s*$</ind:pattern>
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml:45]: Could you also update the rule description, please?

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml [hunk 5]
```diff
@@ -41,4 +60,19 @@
...
+                                id="test_{{{rule_id}}}_default_netstream_rsyslog_dir_rainer" version="1">
+        <ind:object object_ref="obj_{{{rule_id}}}_default_netstream_rsyslog_dir_rainer" />
+    </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_object id="obj_{{{rule_id}}}_default_netstream_rsyslog_dir_rainer"
+                                    comment="Check if DefaultNetstreamDriver is set to gtls in files in /etc/rsyslog.d using Rainer script"
+                                    version="1">
+        <ind:path>/etc/rsyslog.d</ind:path>
+        <ind:filename operation="pattern match">^.*conf$</ind:filename>
+        <ind:pattern operation="pattern match">^\s*global\(.*(?i)\bDefaultNetStreamDriver\b(?-i)="gtls".*\)\s*$</ind:pattern>
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml:75]: Same regex used twice. Could you consider to use a variable, please?

**Review by marcusburghardt — CHANGES_REQUESTED** [review]
Besides the comments in specific files, the remediation for rule `rsyslog_cron_logging` is not working when the `/etc/rsyslog.conf` file is deleted, more specifically in the test scenario  `no_rsyslog_file.fail.sh`. After the remediation, the rsyslog service can't be started. This issue was not introduced by this PR, but was revealed now. Could you take a look on it, please?

**Comment by jan-cerny** [comment]
@marcusburghardt 

 > Besides the comments in specific files, the remediation for rule rsyslog_cron_logging is not working when the /etc/rsyslog.conf file is deleted, more specifically in the test scenario no_rsyslog_file.fail.sh. After the remediation, the rsyslog service can't be started. This issue was not introduced by this PR, but was revealed now. Could you take a look on it, please?

Isn't the test scenario `no_rsyslog_file.fail.sh` stupid? It completely deletes the configuration files. I don't know if we can expect that the remediation for the rule will restore the configuration file and configure all other options that aren't directly mentioned in the rule.

**Comment by marcusburghardt** [comment]
> rsyslog_cron_logging

Yes, I agree this test scenario is very weak. It is fine to remove the files in `/etc/rsyslog.d` but is not fine to remove the whole `rsyslog.conf`. The test scenario `no_cron_logging.fail.sh` seems enough to me.

Perhaps, just to keep a test without rsyslog.d files, we could just rename the `no_rsyslog_file.fail.sh` to `no_rsyslog_d_files.fail.sh` with a content similar to this:

```
#!/bin/bash

rm -rf /etc/rsyslog.d
sed -i '/^[[:space:]]*cron\.\*/d' /etc/rsyslog.conf
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_cron_logging/oval/shared.xml [hunk 6]
```diff
@@ -5,8 +5,12 @@
     <criteria operator="OR">
       <criterion comment="cron is configured in /etc/rsyslog.conf"
       test_ref="test_cron_logging_rsyslog" />
+      <criterion comment="cron is configured in /etc/rsyslog.conf using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_cron_logging/oval/shared.xml:8]:
```suggestion
      <criterion comment="cron is configured in /etc/rsyslog.conf using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_cron_logging/oval/shared.xml [hunk 7]
```diff
@@ -5,8 +5,12 @@
...
+      test_ref="test_cron_logging_rsyslog_rainer" />
       <criterion comment="cron is configured in /etc/rsyslog.d"
       test_ref="test_cron_logging_rsyslog_dir" />
+      <criterion comment="cron is configured in /etc/rsyslog.d using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_cron_logging/oval/shared.xml:12]:
```suggestion
      <criterion comment="cron is configured in /etc/rsyslog.d using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_cron_logging/oval/shared.xml [hunk 8]
```diff
@@ -16,14 +20,29 @@
...
   </ind:textfilecontent54_test>
   <ind:textfilecontent54_object id="obj_cron_logging_rsyslog" version="1">
     <ind:filepath>/etc/rsyslog.conf</ind:filepath>
-    <ind:pattern operation="pattern match">^[\s]*cron\.\*[\s]+/var/log/cron\s*(?:#.*)?$</ind:pattern>
+    <ind:pattern operation="pattern match">{{{ legacy_regex }}}</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist"
+  comment="cron is configured in /etc/rsyslog.conf using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_cron_logging/oval/shared.xml:39]:
```suggestion
  comment="cron is configured in /etc/rsyslog.conf using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_cron_logging/oval/shared.xml [hunk 9]
```diff
@@ -35,7 +54,19 @@
   <ind:textfilecontent54_object id="obj_cron_logging_rsyslog_dir" version="1">
     <ind:path>/etc/rsyslog.d</ind:path>
     <ind:filename operation="pattern match">^.*$</ind:filename>
-    <ind:pattern operation="pattern match">^[\s]*cron\.\*[\s]+/var/log/cron\s*(?:#.*)?$</ind:pattern>
+    <ind:pattern operation="pattern match">{{{ legacy_regex }}}</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_test check="all" check_existence="all_exist"
+  comment="cron is configured in /etc/rsyslog.d using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_cron_logging/oval/shared.xml:62]:
```suggestion
  comment="cron is configured in /etc/rsyslog.d using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_cron_logging/rule.yml [hunk 10]
```diff
@@ -8,7 +8,10 @@ description: |-
     cron job status. If <tt>cron</tt> is not logging to <tt>rsyslog</tt>, it
     can be implemented by adding the following to the <i>RULES</i> section of
     <tt>/etc/rsyslog.conf</tt>:
+    If the legacy syntax is used:
     <pre>cron.*                                                  /var/log/cron</pre>
+    If the modern syntax (Rainer script) is used:
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_cron_logging/rule.yml:13]:
```suggestion
    If the modern syntax (RainerScript) is used:
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml [hunk 11]
```diff
@@ -3,14 +3,21 @@
         {{{ oval_metadata("Rsyslogd must authenticate remote system its sending logs to.") }}}
         <criteria operator="AND">
             <criteria operator="OR">
-                 <criterion comment="Check if $ActionSendStreamDriverAuthMode x509/name is set in /etc/rsyslog.conf"
+                <criterion comment="Check if $ActionSendStreamDriverAuthMode x509/name is set in /etc/rsyslog.conf"
                             test_ref="test_{{{rule_id}}}_action_send_stream_driver_auth_mode" />
+                <criterion comment="Check if StreamDriverAuthMode is set to x509/name in /etc/rsyslog.conf using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml:8]:
```suggestion
                <criterion comment="Check if StreamDriverAuthMode is set to x509/name in /etc/rsyslog.conf using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml [hunk 12]
```diff
@@ -3,14 +3,21 @@
...
+                            test_ref="test_{{{rule_id}}}_action_send_stream_driver_auth_mode_rainer" />
                 <criterion comment="Check if $ActionSendStreamDriverAuthMode x509/name is set in files in /etc/rsyslog.d"
                             test_ref="test_{{{rule_id}}}_action_send_stream_driver_auth_mode_dir" />
+                <criterion comment="Check if StreamDriverAuthMode is set to x509/name in files in /etc/rsyslog.d using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml:12]:
```suggestion
                <criterion comment="Check if StreamDriverAuthMode is set to x509/name in files in /etc/rsyslog.d using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml [hunk 13]
```diff
@@ -22,7 +29,22 @@
                                     comment="Check if  $ActionSendStreamDriverAuthMode x509/name is set in /etc/rsyslog.conf"
                                     version="1">
         <ind:filepath>/etc/rsyslog.conf</ind:filepath>
-        <ind:pattern operation="pattern match">^\$ActionSendStreamDriverAuthMode x509/name$</ind:pattern>
+        <ind:pattern operation="pattern match">{{{ legacy_regex }}}</ind:pattern>
+        <ind:instance datatype="int">1</ind:instance>
+    </ind:textfilecontent54_object>
+
+    <ind:textfilecontent54_test check="all" check_existence="all_exist"
+                                comment="Check if StreamDriverAuthMode is set to x509/name in /etc/rsyslog.conf using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml:37]:
```suggestion
                                comment="Check if StreamDriverAuthMode is set to x509/name in /etc/rsyslog.conf using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml [hunk 14]
```diff
@@ -22,7 +29,22 @@
...
+                                id="test_{{{rule_id}}}_action_send_stream_driver_auth_mode_rainer" version="1">
+
+        <ind:object object_ref="obj_{{{rule_id}}}_action_send_stream_driver_auth_mode_rainer" />
+    </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_object id="obj_{{{rule_id}}}_action_send_stream_driver_auth_mode_rainer"
+                                    comment="Check if StreamDriverAuthMode is set to x509/name in /etc/rsyslog.conf using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml:44]:
```suggestion
                                    comment="Check if StreamDriverAuthMode is set to x509/name in /etc/rsyslog.conf using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml [hunk 15]
```diff
@@ -37,7 +59,23 @@
                                     version="1">
         <ind:path>/etc/rsyslog.d</ind:path>
         <ind:filename operation="pattern match">^.*conf$</ind:filename>
-        <ind:pattern operation="pattern match">^\$ActionSendStreamDriverAuthMode x509/name$</ind:pattern>
+        <ind:pattern operation="pattern match">{{{ legacy_regex }}}</ind:pattern>
         <ind:instance datatype="int">1</ind:instance>
     </ind:textfilecontent54_object>
+
+    <ind:textfilecontent54_test check="all" check_existence="all_exist"
+                                comment="Check if StreamDriverAuthMode is set to x509/name in files in /etc/rsyslog.d using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml:67]:
```suggestion
                                comment="Check if StreamDriverAuthMode is set to x509/name in files in /etc/rsyslog.d using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml [hunk 16]
```diff
@@ -37,7 +59,23 @@
...
+                                id="test_{{{rule_id}}}_action_send_stream_driver_auth_mode_dir_rainer" version="1">
+        <ind:object object_ref="obj_{{{rule_id}}}_action_send_stream_driver_auth_mode_dir_rainer" />
+    </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_object id="obj_{{{rule_id}}}_action_send_stream_driver_auth_mode_dir_rainer"
+                                    comment="Check if StreamDriverAuthMode is set to x509/name in files in /etc/rsyslog.d using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/oval/shared.xml:73]:
```suggestion
                                    comment="Check if StreamDriverAuthMode is set to x509/name in files in /etc/rsyslog.d using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/rule.yml [hunk 17]
```diff
@@ -11,6 +11,11 @@ description: |-
 
     When using <tt>rsyslogd</tt> to off-load logs the remote system must be authenticated.
 
+    Set the following configuration option in /etc/rsyslog.conf or in a file in /etc/rsyslog.d (using legacy syntax):
+    <pre>$ActionSendStreamDriverAuthMode x509/name</pre>
+    Alternatively, use the Rainer script syntax:
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdriverauthmode/rule.yml:16]:
```suggestion
    Alternatively, use the RainerScript syntax:
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml [hunk 18]
```diff
@@ -4,14 +4,21 @@
         {{{ oval_metadata("Rsyslogd must encrypt the off-loading of logs off of the system.") }}}
         <criteria operator="AND">
             <criteria operator="OR">
-                 <criterion comment="Check if $ActionSendStreamDriverMode 1 is set in /etc/rsyslog.conf"
+                <criterion comment="Check if $ActionSendStreamDriverMode 1 is set in /etc/rsyslog.conf"
                             test_ref="test_{{{rule_id}}}_action_send_stream_driver_mode_rsyslog" />
+                <criterion comment="Check if StreamDriverMode is set to 1 in /etc/rsyslog.conf using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml:9]:
```suggestion
                <criterion comment="Check if StreamDriverMode is set to 1 in /etc/rsyslog.conf using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml [hunk 19]
```diff
@@ -4,14 +4,21 @@
...
+                            test_ref="test_{{{rule_id}}}_action_send_stream_driver_mode_rsyslog_rainer" />
                 <criterion comment="Check if $ActionSendStreamDriverMode 1 is set in files in /etc/rsyslog.d"
                             test_ref="test_{{{rule_id}}}_action_send_stream_driver_mode_rsyslog_dir" />
+                <criterion comment="Check if StreamDriverMode is set to 1 in files in /etc/rsyslog.d using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml:13]:
```suggestion
                <criterion comment="Check if StreamDriverMode is set to 1 in files in /etc/rsyslog.d using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml [hunk 20]
```diff
@@ -23,7 +30,22 @@
                                     comment="Check if  $ActionSendStreamDriverMode 1 is set in /etc/rsyslog.conf"
                                     version="1">
         <ind:filepath>/etc/rsyslog.conf</ind:filepath>
-        <ind:pattern operation="pattern match">^\$ActionSendStreamDriverMode 1$</ind:pattern>
+        <ind:pattern operation="pattern match">{{{ legacy_regex }}}</ind:pattern>
+        <ind:instance datatype="int">1</ind:instance>
+    </ind:textfilecontent54_object>
+
+    <ind:textfilecontent54_test check="all" check_existence="all_exist"
+                                comment="Check if StreamDriverMode is set to 1 in /etc/rsyslog.conf using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml:38]:
```suggestion
                                comment="Check if StreamDriverMode is set to 1 in /etc/rsyslog.conf using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml [hunk 21]
```diff
@@ -23,7 +30,22 @@
...
+                                id="test_{{{rule_id}}}_action_send_stream_driver_mode_rsyslog_rainer" version="1">
+
+        <ind:object object_ref="obj_{{{rule_id}}}_action_send_stream_driver_mode_rsyslog_rainer" />
+    </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_object id="obj_{{{rule_id}}}_action_send_stream_driver_mode_rsyslog_rainer"
+                                    comment="Check if StreamDriverMode is set to 1 in /etc/rsyslog.conf using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml:45]:
```suggestion
                                    comment="Check if StreamDriverMode is set to 1 in /etc/rsyslog.conf using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml [hunk 22]
```diff
@@ -38,7 +60,22 @@
                                     version="1">
         <ind:path>/etc/rsyslog.d</ind:path>
         <ind:filename operation="pattern match">^.*conf$</ind:filename>
-        <ind:pattern operation="pattern match">^\$ActionSendStreamDriverMode 1$</ind:pattern>
+        <ind:pattern operation="pattern match">{{{ legacy_regex }}}</ind:pattern>
+        <ind:instance datatype="int">1</ind:instance>
+    </ind:textfilecontent54_object>
+
+    <ind:textfilecontent54_test check="all" check_existence="all_exist"
+                                comment="Check if StreamDriverMode is set to 1 in files in /etc/rsyslog.d using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml:68]:
```suggestion
                                comment="Check if StreamDriverMode is set to 1 in files in /etc/rsyslog.d using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml [hunk 23]
```diff
@@ -38,7 +60,22 @@
...
+                                id="test_{{{rule_id}}}_action_send_stream_driver_mode_rsyslog_dir_rainer" version="1">
+        <ind:object object_ref="obj_{{{rule_id}}}_action_send_stream_driver_mode_rsyslog_dir_rainer" />
+    </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_object id="obj_{{{rule_id}}}_action_send_stream_driver_mode_rsyslog_dir_rainer"
+                                    comment="Check if StreamDriverMode is set to 1 in files in /etc/rsyslog.d using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_actionsendstreamdrivermode/oval/shared.xml:74]:
```suggestion
                                    comment="Check if StreamDriverMode is set to 1 in files in /etc/rsyslog.d using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml [hunk 24]
```diff
@@ -4,14 +4,21 @@
         {{{ oval_metadata("Rsyslogd must encrypt the off-loading of logs off of the system.") }}}
         <criteria operator="AND">
             <criteria operator="OR">
-                 <criterion comment="Check if $DefaultNetstreamDriver gtls is set in /etc/rsyslog.conf"
+                <criterion comment="Check if $DefaultNetstreamDriver gtls is set in /etc/rsyslog.conf"
                             test_ref="test_{{{rule_id}}}_default_netstream_rsyslog" />
+                <criterion comment="Check if DefaultNetstreamDriver is set to gtls in /etc/rsyslog.conf using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml:9]:
```suggestion
                <criterion comment="Check if DefaultNetstreamDriver is set to gtls in /etc/rsyslog.conf using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml [hunk 25]
```diff
@@ -4,14 +4,21 @@
...
+                            test_ref="test_{{{rule_id}}}_default_netstream_rsyslog_rainer" />
                 <criterion comment="Check if $DefaultNetstreamDriver gtls is set in files in /etc/rsyslog.d"
                             test_ref="test_{{{rule_id}}}_default_netstream_rsyslog_dir" />
+                <criterion comment="Check if DefaultNetstreamDriver is set to gtls in files in /etc/rsyslog.d using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml:13]:
```suggestion
                <criterion comment="Check if DefaultNetstreamDriver is set to gtls in files in /etc/rsyslog.d using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml [hunk 26]
```diff
@@ -23,7 +30,22 @@
                                     comment="Check if  $DefaultNetstreamDriver gtls is set in /etc/rsyslog.conf"
                                     version="1">
         <ind:filepath>/etc/rsyslog.conf</ind:filepath>
-        <ind:pattern operation="pattern match">^\$DefaultNetstreamDriver gtls$</ind:pattern>
+        <ind:pattern operation="pattern match">{{{ legacy_regex }}}</ind:pattern>
+        <ind:instance datatype="int">1</ind:instance>
+    </ind:textfilecontent54_object>
+
+    <ind:textfilecontent54_test check="all" check_existence="all_exist"
+                                comment="Check if DefaultNetstreamDriver is set to gtls in /etc/rsyslog.conf using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml:38]:
```suggestion
                                comment="Check if DefaultNetstreamDriver is set to gtls in /etc/rsyslog.conf using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml [hunk 27]
```diff
@@ -23,7 +30,22 @@
...
+                                id="test_{{{rule_id}}}_default_netstream_rsyslog_rainer" version="1">
+
+        <ind:object object_ref="obj_{{{rule_id}}}_default_netstream_rsyslog_rainer" />
+    </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_object id="obj_{{{rule_id}}}_default_netstream_rsyslog_rainer"
+                                    comment="Check if DefaultNetstreamDriver is set to gtls in /etc/rsyslog.conf using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml:45]:
```suggestion
                                    comment="Check if DefaultNetstreamDriver is set to gtls in /etc/rsyslog.conf using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml [hunk 28]
```diff
@@ -38,7 +60,22 @@
                                     version="1">
         <ind:path>/etc/rsyslog.d</ind:path>
         <ind:filename operation="pattern match">^.*conf$</ind:filename>
-        <ind:pattern operation="pattern match">^\$DefaultNetstreamDriver gtls$</ind:pattern>
+        <ind:pattern operation="pattern match">{{{ legacy_regex }}}</ind:pattern>
+        <ind:instance datatype="int">1</ind:instance>
+    </ind:textfilecontent54_object>
+
+    <ind:textfilecontent54_test check="all" check_existence="all_exist"
+                                comment="Check if DefaultNetstreamDriver is set to gtls in files in /etc/rsyslog.d using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml:68]:
```suggestion
                                comment="Check if DefaultNetstreamDriver is set to gtls in files in /etc/rsyslog.d using RainerScript"
```

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml [hunk 29]
```diff
@@ -38,7 +60,22 @@
...
+                                id="test_{{{rule_id}}}_default_netstream_rsyslog_dir_rainer" version="1">
+        <ind:object object_ref="obj_{{{rule_id}}}_default_netstream_rsyslog_dir_rainer" />
+    </ind:textfilecontent54_test>
+
+    <ind:textfilecontent54_object id="obj_{{{rule_id}}}_default_netstream_rsyslog_dir_rainer"
+                                    comment="Check if DefaultNetstreamDriver is set to gtls in files in /etc/rsyslog.d using Rainer script"
```
Comment (marcusburghardt) [inline linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_encrypt_offload_defaultnetstreamdriver/oval/shared.xml:74]:
```suggestion
                                    comment="Check if DefaultNetstreamDriver is set to gtls in files in /etc/rsyslog.d using RainerScript"
```

**Review by marcusburghardt — CHANGES_REQUESTED** [review]
Thanks @jan-cerny . It is much better after using variables for regex. Nice!!
I only suggested some comments to align the name of this rsyslog syntax to the official documentation: RainerScript

Once we solve the test scenario issue I believe we are good to merge.

**Comment by jan-cerny** [comment]
I have changed and renamed the test scenario no_rsyslog_file.fail.sh. I have changed occurrences to RainerScript.

**Review by marcusburghardt — APPROVED** [review]
Nice work. Thanks

**Pull request merged**